### DISCUSSION TITLE
Major macOS upgrade, Monterey -> Ventura

### DISCUSSION THREAD

**Discussion by johnalanwoods** [Homebrew/discussions#4515] [Getting started] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.0.18
ORIGIN: https://github.com/Homebrew/brew
HEAD: b0dc84b117f54f6b18e0528b95477c54e409ebdd
Last commit: 6 days ago
Core tap JSON: 18 May 21:07 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 10
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: 10-core 64-bit arm_firestorm_icestorm
Clang: 14.0.0 build 1400
Git: 2.40.1 => /opt/homebrew/bin/git
Curl: 7.88.1 => /usr/bin/curl
macOS: 13.4-arm64
CLT: N/A
Xcode: 14.2
Rosetta 2: false
```

### Output of `brew doctor`

```shell
`Your system is ready to brew`
```

### Description of issue

I've just updated to Ventura (as I was waiting for some stability). I note that when running `brew update/upgrade` I don't see it go to pull bottles that were specifically compiled for Ventura.

Shouldn't it pull (say for example in the case of [Rust](https://github.com/Homebrew/homebrew-core/blob/master/Formula/rust.rb))

`sha256 cellar: :any,                 arm64_ventura:  "0ca792a1dc62130fce46ebb1fb7cb9adf7d9382eb00a47d444b46f502bd00298"`

To replace the previous binary installed for Monterey?

**Comment by SMillerDev** [comment, marked as answer]
It will on the next update of those

**Reply by johnalanwoods** [reply]
Thank you! So I guess only issue is if existing binaries have any incompatibility?

**Reply by carlocab** [reply]
The only formulae that I'm aware of that have potential incompatibilities are `gcc`, `llvm`, their versioned relatives, and `pkg-config`. The incompatibilities are unlikely to cause you problems though. But if you want to play it safe, do `brew reinstall`.

**Reply by johnalanwoods** [reply]
Thanks both. As always.
